File: app/src/main/java/com/imdc/milkdespencer/roomdb/entities/TransactionEntity.java
package com.imdc.milkdespencer.roomdb.entities;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "transactions")
public class TransactionEntity {

    @PrimaryKey(autoGenerate = true)
    private long id;

    private String userName;
    private String password;
    private String transactionType;
    private String bankTransactionNo;
    private String transactionDate;
    private String transactionTime;
    private double amount;

    private float volume;

    private String milkPrice;

    private String milkTemperature;

    private String machineId;
    private String transactionStatus;
    private String upiId;
    private String uniqueTransactionId;
    private String createdBy;

    // ✅ New Field
    private int uploadToServer = 0;

    public int getUploadToServer() {
        return uploadToServer;
    }

    public void setUploadToServer(int uploadToServer) {
        this.uploadToServer = uploadToServer;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getTransactionType() {
        return transactionType;
    }

    public void setTransactionType(String transactionType) {
        this.transactionType = transactionType;
    }

    public String getBankTransactionNo() {
        return bankTransactionNo;
    }

    public void setBankTransactionNo(String bankTransactionNo) {
        this.bankTransactionNo = bankTransactionNo;
    }

    public String getTransactionDate() {
        return transactionDate;
    }

    public void setTransactionDate(String transactionDate) {
        this.transactionDate = transactionDate;
    }

    public String getTransactionTime() {
        return transactionTime;
    }

    public void setTransactionTime(String transactionTime) {
        this.transactionTime = transactionTime;
    }


    public void setAmount(double amount) {
        this.amount = amount;
    }

    public double getAmount() {
        return amount;
    }

    public float getVolume() {
        return volume;
    }

    public void setVolume(float volume) {
        this.volume = volume;
    }

    public String getMilkPrice() {
        return milkPrice;
    }

    public void setMilkPrice(String milkPrice) {
        this.milkPrice = milkPrice;
    }

    public String getMilkTemperature() {
        return milkTemperature;
    }

    public void setMilkTemperature(String milkTemperature) {
        this.milkTemperature = milkTemperature;
    }

    public String getMachineId() {
        return machineId;
    }

    public void setMachineId(String machineId) {
        this.machineId = machineId;
    }

    public String getTransactionStatus() {
        return transactionStatus;
    }

    public void setTransactionStatus(String transactionStatus) {
        this.transactionStatus = transactionStatus;
    }

    public String getUpiId() {
        return upiId;
    }

    public void setUpiId(String upiId) {
        this.upiId = upiId;
    }

    public String getUniqueTransactionId() {
        return uniqueTransactionId;
    }

    public void setUniqueTransactionId(String uniqueTransactionId) {
        this.uniqueTransactionId = uniqueTransactionId;
    }

    public String getCreatedBy() {
        return createdBy;
    }

    public void setCreatedBy(String createdBy) {
        this.createdBy = createdBy;
    }
}

File: app/src/main/java/com/imdc/milkdespencer/roomdb/entities/TransactionEntity.kt
package com.imdc.milkdespencer.roomdb.entities

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "transactions")
class TransactionEntity {
    @JvmField
    @PrimaryKey(autoGenerate = true)
    var id: Long = 0
    @JvmField
    var userName: String? = null
    @JvmField
    var password: String? = null
    @JvmField
    var transactionType: String? = null
    @JvmField
    var bankTransactionNo: String? = null
    @JvmField
    var transactionDate: String? = null
    @JvmField
    var transactionTime: String? = null
    @JvmField
    var amount = 0.0
    @JvmField
    var volume = 0f
    @JvmField
    var milkPrice: String? = null
    @JvmField
    var milkTemperature: String? = null
    @JvmField
    var machineId: String? = null
    @JvmField
    var transactionStatus: String? = null
    @JvmField
    var upiId: String? = null
    @JvmField
    var uniqueTransactionId: String? = null
    @JvmField
    var createdBy: String? = null
}
